src/poetry/installation/executor.py

```diff
@@ -769,6 +772,8 @@ def _create_git_url_reference(
                 "commit_id": package.source_resolved_reference,
             },
         }
+        if package.source_subdirectory:
+            reference["subdirectory"] = package.source_subdirectory
```

Comment:
And this needs to be read back in `InstalledRepository.create_package_from_pep610`.

```diff
         package = Package(
             distribution.metadata["name"],
             distribution.metadata["version"],
             source_type=source_type,
             source_url=source_url,
             source_reference=source_reference,
             source_resolved_reference=source_resolved_reference,
+            source_subdirectory=url_reference.get("subdirectory"),
             develop=develop,
         )
```

(`InstalledRepository.create_package_from_distribution` may also need some kind of update?)